clang/lib/Format/TokenAnnotator.cpp

```diff
@@ -3122,6 +3122,9 @@ class AnnotatingParser {
       }
     }
 
+    if (PrevToken->isTypeName(LangOpts))
```

Comment:
So this does not work with `bool foo = requires { static_cast<MyType &&>(1); };`?

Reply:
It works if `TypeNames: [MyType]` is added to the config, but I've added a heuristic so that it won't be necessary.